internal/server/commonpool.go

```diff
@@ -0,0 +1,63 @@
+package server
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+
+//islandDistribute distributes resources to island as requested 
+func (s *SOMASServer) islandDistribute(resource int, islandID shared.ClientID) { 
+
+	resChan := make(chan clientInfoUpdateResult, 1)
+
+	ci := s.gameState.ClientInfos[islandID]
+	ci.Resources += resource 	
+	resChan <- clientInfoUpdateResult{
+		ID:  islandID,
+		Ci:  ci,
+		Err: nil,
+	}			
+	s.gameState.ClientInfos[islandID] = ci
+	s.gameState.Environment.CommonPool.Resource -= uint(resource)
```

Comment:
Redundant concurrency, the boss will speak with you shortly.

Reply:
![image](https://user-images.githubusercontent.com/33488131/103105316-f817ac00-4624-11eb-9a08-9e54c59422a4.png)